### PR TITLE
Apply space_before_type_annotation to struct ctors (#1109)

### PR DESCRIPTION
**Pull request by HighCommander4**

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -1661,13 +1661,18 @@ fn rewrite_field(context: &RewriteContext,
                  offset: Indent)
                  -> Option<String> {
     let name = &field.ident.node.to_string();
-    let overhead = name.len() + 2;
+    let separator = if context.config.space_before_type_annotation {
+        " : "
+    } else {
+        ": "
+    };
```
Comment (nrc) [inline src/expr.rs]: Can this be factored out into a function, I think you use the same idiom in the last PR?

**Comment by nrc** [comment]
Looks good, I just had one comment, r+ with that addressed

**Comment by HighCommander4** [comment]
Addressed comment in the updated version.

**Comment by nrc** [comment]
Thank you!

**Pull request merged**